### PR TITLE
Implementing REST wrapper for keycloak legos

### PR DESCRIPTION
**Pull request by jayasimha-raghavan-unskript**
## Description
Please include a summary of the change, motivation and context.

* Implemented REST based API calls for keycloak Actions.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

#### Health Check 
```
root@awesome-awesome-runbooks-0:~/test# python ./keycloak_get_service_health.py
Keycloak Service is Healthy. Master realm is available.
root@awesome-awesome-runbooks-0:~/test# 
```

#### Audit Report
```
root@awesome-awesome-runbooks-0:~/test# python ./keycloak_get_audit_report.py
+---------------------+--------+--------------------------------------+-------------+----------------+---------+
| Time                | Type   | User ID                              | Client ID   | IP Address     | Error   |
+=====================+========+======================================+=============+================+=========+
| 2024-02-01 17:23:47 | LOGIN  | 747d6b08-52ae-4c9f-8fd5-a715ca05da5f | admin-cli   | 192.168.144.36 |         |
+---------------------+--------+--------------------------------------+-------------+----------------+---------+
| 2024-02-01 17:20:48 | LOGIN  | 747d6b08-52ae-4c9f-8fd5-a715ca05da5f | admin-cli   | 192.168.144.36 |         |
+---------------------+--------+--------------------------------------+-------------+----------------+---------+
| 2024-02-01 17:20:23 | LOGIN  | 747d6b08-52ae-4c9f-8fd5-a715ca05da5f | admin-cli   | 192.168.144.36 |         |
+---------------------+--------+--------------------------------------+-------------+----------------+---------+
| 2024-02-01 17:17:53 | LOGIN  | 747d6b08-52ae-4c9f-8fd5-a715ca05da5f | admin-cli   | 192.168.144.36 |         |
+---------------------+--------+--------------------------------------+-------------+----------------+---------+
| 2024-02-01 17:12:36 | LOGIN  | 747d6b08-52ae-4c9f-8fd5-a715ca05da5f | admin-cli   | 192.168.144.36 |         |
+---------------------+--------+--------------------------------------+-------------+----------------+---------+
| 2024-02-01 17:11:12 | LOGIN  | 747d6b08-52ae-4c9f-8fd5-a715ca05da5f | admin-cli   | 192.168.144.36 |         |
+---------------------+--------+--------------------------------------+-------------+----------------+---------+
| 2024-02-01 17:04:04 | LOGIN  | 747d6b08-52ae-4c9f-8fd5-a715ca05da5f | admin-cli   | 192.168.144.36 |         |
+---------------------+--------+--------------------------------------+-------------+----------------+---------+
| 2024-02-01 17:03:32 | LOGIN  | 747d6b08-52ae-4c9f-8fd5-a715ca05da5f | admin-cli   | 192.168.144.36 |         |
+---------------------+--------+--------------------------------------+-------------+----------------+---------+
| 2024-02-01 13:15:03 | LOGIN  | 747d6b08-52ae-4c9f-8fd5-a715ca05da5f | admin-cli   | 192.168.144.36 |         |
+---------------------+--------+--------------------------------------+-------------+----------------+---------+
| 2024-02-01 13:13:07 | LOGIN  | 747d6b08-52ae-4c9f-8fd5-a715ca05da5f | admin-cli   | 192.168.144.36 |         |
+---------------------+--------+--------------------------------------+-------------+----------------+---------+

```

### Checklist:
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository.

### CODE REVIEW

File: Keycloak/legos/keycloak_get_service_health/keycloak_get_service_health.py [hunk 1]
```diff
@@ -29,7 +31,34 @@ def keycloak_get_service_health(handle) -> Tuple:
...
+            raise e
+        
+        if not isinstance(handle, UnskriptKeycloakWrapper):
+            raise ValueError(f"Unable to Find Keycloak Package! {e}")
+        
+        access_token = get_keycloak_token(handle)
+        realms_url = f"{handle.server_url}/admin/realms"
+        headers = {"Authorization": f"Bearer {access_token}"}
+        response = requests.get(realms_url, headers=headers)
+        response.raise_for_status()
```
Comment (amit-chandak-unskript) [inline Keycloak/legos/keycloak_get_service_health/keycloak_get_service_health.py:49]: This is not behind a try/catch

File: Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py [hunk 2]
```diff
@@ -48,6 +58,25 @@ def keycloak_get_audit_report(handle) -> List:
...
+        except:
+            raise e
+        
+        if not isinstance(handle, UnskriptKeycloakWrapper):
+            raise ValueError(f"Unable to Find Keycloak Package! {e}")
+        access_token = get_keycloak_token(handle)
+        events_url = f"{handle.server_url}/admin/realms/{handle.realm_name}/events"
+        headers = {"Authorization": f"Bearer {access_token}"}
+
+        response = requests.get(events_url, headers=headers)
```
Comment (amit-chandak-unskript) [inline Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py:75]: Need try/catch

File: Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py [hunk 3]
```diff
@@ -17,20 +20,27 @@ def keycloak_get_audit_report_printer(output):
...
-            "Time": event['time'],
-            "Type": event['type'],
-            "User ID": event['userId'],
-            "Client ID": event['clientId'],
-            "IP Address": event['ipAddress'],
-            "Error": event.get('error', '')
-        })
+        time = event.get('time') if event.get('time') else ''
+        _type = event.get('type') if event.get('type') else \
+               event.get('operationType') if event.get('operationType') else ''
```
Comment (amit-chandak-unskript) [inline Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py:27]: Use default value wherever possible

File: Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py [hunk 4]
```diff
@@ -48,6 +58,25 @@ def keycloak_get_audit_report(handle) -> List:
         return events if events else []
 
     except Exception as e:
-        raise e
+        try:
```
Comment (amit-chandak-unskript) [inline Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py:61]: Keep it simple, assume we are using request based keycloak connector

File: Keycloak/legos/keycloak_get_service_health/keycloak_get_service_health.py (hunk 1 above)
Comment (jayasimha-raghavan-unskript) [reply Keycloak/legos/keycloak_get_service_health/keycloak_get_service_health.py:35]: Done.

File: Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py (hunk 2 above)
Comment (jayasimha-raghavan-unskript) [reply Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py:67]: Done

File: Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py (hunk 3 above)
Comment (jayasimha-raghavan-unskript) [reply Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py:27]: Done

File: Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py (hunk 4 above)
Comment (jayasimha-raghavan-unskript) [reply Keycloak/legos/keycloak_get_audit_report/keycloak_get_audit_report.py]: Done

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**